__tests__/hathorwallet.test.js

```diff
@@ -32,5 +43,133 @@ test('Protected xpub wallet methods', async () => {
+  hWallet.getAddressIndex = () => 1;
+
+  let returnedTx = hWallet.getSignatures(tx0);
+  expect(returnedTx).toBe(tx0);
+  // The transaction is filled with the input data
+  expect(tx0.inputs[0].data).not.toBe(null);
+  expect(tx0.inputs[1].data).toBe(null);
+  expect(tx0.inputs[2].data).not.toBe(null);
+
+  // Mocks expectations
+  expect(mockPrivkey).toBeCalledWith('mocked-encrypted-privkey', '123');
+  expect(mockPrivkey).toBeCalledTimes(1);
+  // Only 2 inputs are from our fake wallet
+  expect(mockInputData).toBeCalledTimes(2);
+  // And all inputs are made with valid signatures
+  mockInputData.mock.calls.forEach(args => {
+    const [sigDER, pubkey] = args;
+    expect(crypto.ECDSA.verify(
+      tx0.getDataToSignHash(),
+      crypto.Signature.fromDER(sigDER),
+      PublicKey.fromBuffer(pubkey),
+      'little',
+    )).toBe(true);
+  });
+  // We set the inputData with the correct value
+  expect(
+    mockSetData.mock.calls[0][0].toString('hex')
+  ).toEqual(mockInputData.mock.results[0].value.toString('hex'));
+  expect(
+    mockSetData.mock.calls[1][0].toString('hex')
+  ).toEqual(mockInputData.mock.results[1].value.toString('hex'));
+
+  // Clear mocks
+  mockPrivkey.mockClear();
+  mockInputData.mockClear();
+  mockSetData.mockClear();
+
+  const tx1 = new Transaction(
+    [new Input(inputTx, 0), new Input(inputTx, 1), new Input(inputTx, 2)],
+    outputs,
+    { version: DEFAULT_TX_VERSION, tokens: [tokenUid] },
+  );
+  // Calling with a pin code
+  returnedTx = hWallet.getSignatures(tx1, { pinCode: 'another-PIN' });
+  expect(returnedTx).toBe(tx1);
+  // The transaction is filled with the input data like before
+  expect(tx1.inputs[0].data).not.toBe(null);
+  expect(tx1.inputs[1].data).toBe(null);
+  expect(tx1.inputs[2].data).not.toBe(null);
+
+  // Mocks expectations
+  expect(mockPrivkey).toBeCalledWith('mocked-encrypted-privkey', 'another-PIN');
+  expect(mockPrivkey).toBeCalledTimes(1);
+  // Only 2 inputs are from our fake wallet
+  expect(mockInputData).toBeCalledTimes(2);
+  // And all inputs are made with valid signatures
+  mockInputData.mock.calls.forEach(args => {
+    const [sigDER, pubkey] = args;
+    expect(crypto.ECDSA.verify(
+      tx1.getDataToSignHash(),
+      crypto.Signature.fromDER(sigDER),
+      PublicKey.fromBuffer(pubkey),
+      'little',
+    )).toBe(true);
+  });
+  // We set the inputData with the correct value
+  expect(
+    mockSetData.mock.calls[0][0].toString('hex')
+  ).toEqual(mockInputData.mock.results[0].value.toString('hex'));
+  expect(
+    mockSetData.mock.calls[1][0].toString('hex')
+  ).toEqual(mockInputData.mock.results[1].value.toString('hex'));
+
+  // Calling without pin should throw an error
+  hWallet.pinCode = null;
+  expect(() => {
+   return hWallet.getSignatures(tx0);
+  }).toThrow('Pin is required.');
+
+  // Cleanup
+  mockPrivkey.mockRestore();
+  mockInputData.mockRestore();
+  mockSetData.mockRestore();
+  mockStorageGet.mockRestore();
 });
```

Comment:
Github is informing that this file does not end with a newline.
Is our linter configured to also get that? Maybe it would be a good idea to add this rule too.